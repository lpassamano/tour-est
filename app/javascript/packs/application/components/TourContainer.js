import React, { Component } from "react";
import PropTypes from "prop-types";
import Tour from "./Tour";
import AddPointForm from "./AddPointForm";
import PointsList from "./PointsList";

export class TourContainer extends Component {
  state = {
    isShowingPointForm: false
  };

  showPointForm = () => {
    this.setState({ isShowingPointForm: true });
  };

  hidePointForm = () => {
    this.setState({ isShowingPointForm: false });
  };

  render() {
    return (
      <div>
        <Tour tourId={this.props.tourId} />
        {this.state.isShowingPointForm ? (
          <AddPointForm
            onHide={this.hidePointForm}
            onShow={this.showPointForm}
            tourId={this.props.tourId}
          />
        ) : (
          <div>
            <button type="button" onClick={this.showPointForm}>
              Add Point to Tour
            </button>
            <h3>Points on tour</h3>
            <PointsList tourId={this.props.tourId} />
          </div>
        )}
      </div>
    );
  }
}

TourContainer.propTypes = {
  tourId: PropTypes.string
};

export default TourContainer;
