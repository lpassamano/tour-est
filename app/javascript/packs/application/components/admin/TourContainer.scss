.tour-details {
  padding-bottom: 48px;
}

h3.point-header {
  padding-bottom: 16px;
}
