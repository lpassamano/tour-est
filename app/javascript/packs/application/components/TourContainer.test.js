import { TourContainer } from "./TourContainer";
import React from "react";
import { shallow } from "enzyme";

describe("<TourContainer />", () => {
  test("render", () => {
    const tour = {
      title: "Philly Tour",
      estimated_time: "2 hours",
      description: "Walking tour of Philly's architecture and public art",
      starting_point: "City Hall",
      directions: "Meet at the West entrance"
    };
    const getTour = jest.fn();
    const component = shallow(<TourContainer tour={tour} getTour={getTour} />);
    expect(component).toMatchSnapshot();
  });
});
