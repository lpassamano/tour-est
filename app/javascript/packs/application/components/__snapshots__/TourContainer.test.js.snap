// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TourContainer /> render 1`] = `
<div>
  <Connect(Tour)
    tourId="1"
  />
  <h3
    className="point-header"
  >
    What you will see on this tour:
  </h3>
  <Connect(PointCards)
    tourId="1"
  />
</div>
`;
